patches/ctypes_init.diff

```diff
@@ -14,11 +13,36 @@
 +    's390x': 's390x-linux-gnu',
 +}
 +
++
+ class CDLL(object):
+     """An instance of this class represents a loaded dll/shared
+     library, exporting functions using the standard C calling
+@@ -344,7 +354,15 @@
+         self._FuncPtr = _FuncPtr
+ 
+         if handle is None:
+-            self._handle = _dlopen(self._name, mode)
++            name = self._name
++            if name is not None and _os.getenv('SNAP_NAME', '') == 'snapcraft':
```

Comment:
It would be good to document why this indirection is needed

Reply:
ah, that is legacy, but by default cpython loads only what is in ld cache, so it wouldn't pick up any library inside the snap, bad thing for classic :wink: